### PR TITLE
fix(rpc): return app version from state store for status endpoint

### PR DESCRIPTION
**Pull request by rootulp**
Fixes https://github.com/celestiaorg/celestia-app/issues/3378

## Testing

On a local node that upgraded from v1 -> v2: 

```
{"jsonrpc":"2.0","id":-1,"result":{"node_info":{"protocol_version":{"p2p":"8","block":"11","app":"2"}
```

### CODE REVIEW

**Review by rach-id — APPROVED** [review]

File: rpc/core/status_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			nodeInfo := core.GetNodeInfo(tc.env, tc.consensusParams)
+			assert.Equal(t, tc.want, nodeInfo.ProtocolVersion.App)
+		})
+	}
+}
+
+// transport is copy + pasted from core because it isn't exported.
```
Comment (staheri14) [inline rpc/core/status_test.go:49]: [nit] A link to where this is copied from would be useful for future references.

File: rpc/core/status.go [hunk 2]
```diff
@@ -85,3 +91,15 @@ func validatorAtHeight(h int64) *types.Validator {
 	_, val := vals.GetByAddress(privValAddress)
 	return val
 }
+
+func GetNodeInfo(env *Environment, consensusParams cmtproto.ConsensusParams) p2p.DefaultNodeInfo {
```
Comment (staheri14) [inline rpc/core/status.go:95]: [suggestion] Would it make sense to move the retrieval of the consensus state inside this function to make it self-sufficient?
```go
func GetNodeInfo(env *Environment) (*p2p.DefaultNodeInfo, error) {
consensusParams, err := env.StateStore.LoadConsensusParams(latestHeight)
	if err != nil {
		return nil, err
	}
...
```

**Review by staheri14 — APPROVED** [review]
LGTM! just added a few non-blocking suggestions.

**Review by evan-forbes — DISMISSED** [review]

File: rpc/core/status.go (hunk 2 above)
Comment (rootulp) [reply rpc/core/status.go:95]: Yea can do!

Comment (rootulp) [reply rpc/core/status.go]: Done!  f936054

**Review by staheri14 — APPROVED** [review]
Thanks for addressing my comment! LGTM!

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**